ansible_ai_connect/ai/api/views.py

```diff
@@ -1012,3 +1016,77 @@ def write_to_segment(
                 },
             )
         send_segment_event(event, "codegenPlaybook", user)
+
+
+class Chat(APIView):
+    """
+    Send a message to the backend chatbot service and get a reply.
+    """
+
+    permission_classes = [
+        permissions.IsAuthenticated,
+        IsAuthenticatedOrTokenHasScope,
+    ]
+    required_scopes = ["read", "write"]
+
+    CHAT_API_NOT_ENABLED = "Chat API is not enabled."
+    INVALID_CHAT_REQUEST = "An invalid chat request was received."
+    INVALID_CHAT_RESPONSE = "An invalid chat response was received."
+
+    def __init__(self):
+        self.chat_enabled = (
+            settings.CHATBOT_URL
+            and settings.CHATBOT_DEFAULT_MODEL
+            and settings.CHATBOT_DEFAULT_PROVIDER
+        )
+
+    @extend_schema(
+        request=ChatRequestSerializer,
+        responses={
+            200: ChatResponseSerializer,
+            400: OpenApiResponse(description="Bad Request"),
+        },
+        summary="Chat request",
+    )
+    def post(self, request) -> Response:
+        if not self.chat_enabled:
```

Comment:
You could probably do this in the `__init__`? If it's not enabled we can fail fast.

Reply:
That's true.  I'll updat the code. Thanks.

Reply:
I tried to move the code to `__init__`.  However, the code was called from CSRF protection code and did not work well. I reverted that and kept the original logic.